test/e2e/common.go

```diff
@@ -274,7 +274,7 @@ func EnsureControlPlaneInitialized(ctx context.Context, input clusterctl.ApplyCl
 	}, input.WaitForControlPlaneIntervals...).Should(Succeed(), "API Server was not reachable in time")
 
 	_, hasWindows := cluster.Labels["cni-windows"]
-	if kubeadmControlPlane.Spec.KubeadmConfigSpec.ClusterConfiguration.ControllerManager.ExtraArgs["cloud-provider"] == "external" {
+	if kubeadmControlPlane.Spec.KubeadmConfigSpec.ClusterConfiguration.ControllerManager.ExtraArgs["cloud-provider"] != "azure" {
```

Comment:
interesting choice here, I assume there's no functional change, was this just to make this condition look more like "we always do this unless this" instead of "only do this sometimes if this is true"?

Reply:
yes, this is to future proof this code for 1.27 when cloud-provider arg will become optional